File: ios/MenCare/Features/Staff/StaffBookingScreen.swift
import SwiftUI
import Observation

/// Same flow as the admin manual booking, but scoped to the signed-in staff
/// member's own chair: no operator picker, and the service list is limited to
/// what that operator actually performs (so slot generation never comes back
/// empty because of an eligibility mismatch the staff member can't see).
@MainActor
@Observable
final class StaffBookingViewModel {

    private let auth: AuthRepository
    private let booking: BookingRepository
    private let crm: CrmRepository
    private let catalog: CatalogRepository

    var query = ""
    var results: [ClientRecord] = []
    var selectedClient: ClientRecord?
    var creatingClient = false
    var newFirst = ""
    var newLast = ""
    var newPhone = ""
    var newClientErrors: [String: String] = [:]
    var newClientError: String?
    var creatingClientInFlight = false
    var date: LocalDate = .today()
    var selectedServiceIds: [String] = []
    var slots: [LocalTime] = []
    var selectedSlot: LocalTime?
    /// Time the sheet was opened on (a tap on the agenda) or last picked:
    /// selected again as soon as the chosen services fit there.
    var preferredSlot: LocalTime?
    var sendSms = true
    var done = false

    @ObservationIgnored private var searchTask: Task<Void, Never>?
    /// "Modifica" dal dettaglio appuntamento: l'appuntamento che questo modulo sostituisce.
    @ObservationIgnored private let editingId: String?

    var isEditing: Bool { editingId != nil }

    init(
        auth: AuthRepository, booking: BookingRepository, catalog: CatalogRepository, crm: CrmRepository,
        date: LocalDate = .today(), time: LocalTime? = nil,
        editAppointment: Appointment? = nil, editClient: ClientRecord? = nil
    ) {
        self.auth = auth
        self.booking = booking
        self.catalog = catalog
        self.crm = crm
        if let editAppointment {
            editingId = editAppointment.id
            self.date = editAppointment.date
            selectedServiceIds = editAppointment.serviceIds
            preferredSlot = editAppointment.time
            selectedClient = editClient
            query = editClient?.fullName ?? ""
        } else {
            editingId = nil
            self.date = date
            preferredSlot = time
        }
    }

    let state = LoadState()
    private(set) var op: Operator?
    /// Solo i servizi che l'operatore esegue davvero: altrimenti la griglia
    /// tornerebbe vuota per un'incompatibilita' che lui non puo' vedere.
    private(set) var services: [Service] = []
    var saveError: String?

    func load() async {
        await state.run {
            let operatorId = try await auth.currentUser()?.operatorId
            let snapshot = try await catalog.catalog()
            op = snapshot.operators.first { $0.id == operatorId }
            services = snapshot.services.filter { op?.serviceIds.contains($0.id) ?? false }
        }
        refreshSlots()
    }

    var totalMinutes: Int {
        services.filter { selectedServiceIds.contains($0.id) }.reduce(0) { $0 + $1.durationMinutes }
    }

    var canSave: Bool { selectedClient != nil && !selectedServiceIds.isEmpty && selectedSlot != nil }

    /// The preferred time is taken for the chosen services.
    var preferredUnavailable: Bool {
        guard let preferredSlot, !selectedServiceIds.isEmpty else { return false }
        return selectedSlot == nil && !slots.contains(preferredSlot)
    }

    func search(_ text: String) {
        query = text
        searchTask?.cancel()
        guard !text.trimmingCharacters(in: .whitespaces).isEmpty else {
            results = []
            return
        }
        // La ricerca la fa il server: si aspetta la pausa fra un tasto e l'altro.
        searchTask = Task {
            try? await Task.sleep(for: .milliseconds(300))
            guard !Task.isCancelled else { return }
            results = Array(((try? await crm.clients(query: text, segment: .tutti)) ?? []).prefix(5))
        }
    }

    func selectClient(_ client: ClientRecord) {
        selectedClient = client
        results = []
        query = client.fullName
        creatingClient = false
    }

    func startCreateClient() {
        creatingClient = true
        selectedClient = nil
    }

    func newClientFieldChanged() {
        newClientError = nil
        if !newClientErrors.isEmpty {
            newClientErrors = validateNewClient(firstName: newFirst, lastName: newLast, phone: newPhone)
        }
    }

    func createClient() {
        guard !creatingClientInFlight else { return }
        let errors = validateNewClient(firstName: newFirst, lastName: newLast, phone: newPhone)
        if !errors.isEmpty {
            // Prima il fallimento veniva ingoiato e il tasto sembrava rotto.
            newClientErrors = errors
            newClientError = L("validation_form_invalid")
            return
        }
        newClientErrors = [:]
        newClientError = nil
        creatingClientInFlight = true
        let phoneToSave = normalizePhone(newPhone) ?? newPhone
        Task {
            let result = await crm.createClient(
                firstName: newFirst.trimmingCharacters(in: .whitespaces),
                lastName: newLast.trimmingCharacters(in: .whitespaces),
                phone: phoneToSave
            )
            creatingClientInFlight = false
            switch result {
            case .success(let record):
                selectClient(record)
            case .failure:
                newClientError = L("validation_save_failed")
            }
        }
    }

    func selectDate(_ newDate: LocalDate) {
        date = newDate
        selectedSlot = nil
        refreshSlots()
    }

    func toggleService(_ serviceId: String) {
        if let index = selectedServiceIds.firstIndex(of: serviceId) {
            selectedServiceIds.remove(at: index)
        } else {
            selectedServiceIds.append(serviceId)
        }
        selectedSlot = nil
        refreshSlots()
    }

    func selectSlot(_ slot: LocalTime) {
        selectedSlot = slot
        preferredSlot = slot
    }

    func refreshSlots() {
        Task {
            guard let opId = op?.id, !selectedServiceIds.isEmpty else {
                slots = []
                return
            }
            // In modifica l'appuntamento non occupa il proprio posto: il suo orario resta sceglibile.
            slots = (try? await booking.availability(
                operatorId: opId, serviceIds: selectedServiceIds, date: date,
                ignoreAppointmentId: editingId
            ).slots) ?? []
            if selectedSlot == nil, let preferredSlot, slots.contains(preferredSlot) {
                selectedSlot = preferredSlot
            }
        }
    }

    func save() {
        guard let client = selectedClient, let slot = selectedSlot, let opId = op?.id else { return }
        Task {
            saveError = nil
            let result = await booking.book(
                BookingRequest(
                    clientId: client.id, operatorId: opId, serviceIds: selectedServiceIds,
                    start: date.atTime(slot), channel: .phone,
                    // Modifica: il server sostituisce l'originale nella stessa transazione.
                    replacesAppointmentId: editingId
                )
            )
            switch result {
            case .success:
                done = true
            case .failure(let error):
                saveError = error.displayMessage
                // Se l'orario e' stato preso nel frattempo la griglia va rifatta.
                refreshSlots()
            }
        }
    }
}

/// Staff self-service booking sheet: same flow as the admin manual booking,
/// but the operator is always "me" — lets staff take a phone/walk-in booking
/// without the titolare.
struct StaffBookingScreen: View {
    @Bindable var viewModel: StaffBookingViewModel
    let onDismiss: () -> Void

    @State private var showDatePicker = false
    @State private var headerHeight: CGFloat = 0
    @State private var formHeight: CGFloat = 0
    @State private var footerHeight: CGFloat = 0

    var body: some View {
        VStack(spacing: 0) {
            HStack {
                BarAction(text: L("staff_booking_cancel"), action: onDismiss, color: .ink)
                    .frame(width: 72, alignment: .leading)
                Text(L(viewModel.isEditing ? "staff_booking_title_edit" : "staff_booking_title"))
                    .font(Typo.cormorant(21, weight: .regular))
                    .foregroundStyle(Color.ink)
                    .lineLimit(1)
                    .frame(maxWidth: .infinity)
                BarAction(text: L("staff_booking_save"), action: viewModel.save, color: .oliveLight)
                    .frame(width: 72, alignment: .trailing)
            }
            .padding(.horizontal, 20)
            .padding(.vertical, 6)
            .measureHeight($headerHeight)

            ScrollView {
                VStack(alignment: .leading, spacing: 12) {
                    BrandSectionLabel(text: L("staff_booking_client"))
                    // In modifica il cliente è quello dell'appuntamento: si mostra e basta.
                    if viewModel.isEditing {
                        FixedClientRow(client: viewModel.selectedClient)
                    } else {
                        FilledTextField(
                            label: "",
                            text: Binding(get: { viewModel.query }, set: { viewModel.search($0) }),
                            placeholder: L("staff_booking_search_hint"),
                            leadingSystemImage: "magnifyingglass"
                        )
                        clientList
                    }

                    if viewModel.creatingClient {
                        NewClientForm(
                            firstLabel: L("staff_booking_new_client_first"),
                            lastLabel: L("staff_booking_new_client_last"),
                            phoneLabel: L("staff_booking_new_client_phone"),
                            ctaLabel: L("staff_booking_new_client"),
                            firstName: $viewModel.newFirst,
                            lastName: $viewModel.newLast,
                            phone: $viewModel.newPhone,
                            errors: viewModel.newClientErrors,
                            formError: viewModel.newClientError,
                            creating: viewModel.creatingClientInFlight,
                            onCreate: viewModel.createClient
                        )
                        .onChange(of: viewModel.newFirst) { viewModel.newClientFieldChanged() }
                        .onChange(of: viewModel.newLast) { viewModel.newClientFieldChanged() }
                        .onChange(of: viewModel.newPhone) { viewModel.newClientFieldChanged() }
                    }

                    BrandSectionLabel(text: L("staff_booking_date"))
                    Button {
                        showDatePicker = true
                    } label: {
                        HStack {
                            Text(formatDateShort(viewModel.date))
                                .font(Typo.bodyLarge)
                                .foregroundStyle(Color.ink)
                            Spacer()
                            Image(systemName: "chevron.down")
                                .font(.system(size: 13))
                                .foregroundStyle(Color.textMuted)
                        }
                        .padding(.horizontal, 14)
                        .frame(height: 56)
                        .background(RoundedRectangle(cornerRadius: Radii.md).fill(Color.bone))
                        .overlay(
                            RoundedRectangle(cornerRadius: Radii.md)
                                .strokeBorder(Color.stoneBorder, lineWidth: 1.5)
                        )
                    }
                    .buttonStyle(.plain)

                    BrandSectionLabel(text: L("staff_booking_services"))
                    // Stessa tendina multi-selezione della prenotazione del titolare.
                    MultiSelectDropdown(
                        options: viewModel.services.map { DropdownOption(id: $0.id, label: $0.name) },
                        selectedIds: viewModel.selectedServiceIds,
                        onToggle: viewModel.toggleService,
                        placeholder: L("manual_services_placeholder_staff")
                    )

                    BrandSectionLabel(text: L("staff_booking_slots", formatDuration(viewModel.totalMinutes)))
                    SlotChipRow(
                        slots: viewModel.slots,
                        selected: viewModel.selectedSlot,
                        onSelect: viewModel.selectSlot,
                        label: formatTime,
                        emptyLabel: viewModel.preferredSlot.map { L("staff_booking_slots_empty_preferred", formatTime($0)) }
                            ?? L("staff_booking_slots_empty")
                    )
                    if viewModel.preferredUnavailable, let preferred = viewModel.preferredSlot {
                        Text(L("staff_booking_slot_unavailable", formatTime(preferred)))
                            .font(Typo.jost(12))
                            .foregroundStyle(Color.textMuted)
                    }

                    HStack {
                        VStack(alignment: .leading, spacing: 0) {
                            Text(L("staff_booking_sms"))
                                .font(Typo.jost(15, weight: .medium))
                                .foregroundStyle(Color.ink)
                            Text(L("staff_booking_sms_hint"))
                                .font(Typo.jost(12))
                                .foregroundStyle(Color.textMuted)
                        }
                        Spacer()
                        BrandSwitch(isOn: $viewModel.sendSms)
                    }
                    .padding(.horizontal, 16)
                    .padding(.vertical, 12)
                    .background(RoundedRectangle(cornerRadius: Radii.md).fill(Color.bone))
                    .overlay(
                        RoundedRectangle(cornerRadius: Radii.md)
                            .strokeBorder(Color.stoneBorder, lineWidth: 1.5)
                    )

                }
                .padding(.horizontal, 20)
                .padding(.top, 12)
                .padding(.bottom, 12)
                .measureHeight($formHeight)
            }
            .scrollBounceBehavior(.basedOnSize)
            // Fuori dallo scroll: il pulsante resta sempre a vista, subito sotto il modulo.
            VStack(spacing: 8) {
                FormErrorBanner(message: viewModel.saveError)
                AccentButton(
                    text: L(viewModel.isEditing ? "staff_booking_cta_edit" : "staff_booking_cta"),
                    action: viewModel.save,
                    enabled: viewModel.canSave,
                    height: 56,
                    corner: 16
                )
            }
            .padding(.horizontal, 20)
            .padding(.top, 4)
            .padding(.bottom, 12)
            .measureHeight($footerHeight)
        }
        .background(Color.bone)
        .task { await viewModel.load() }
        // Alta quanto il contenuto, come "Ferie e permessi": niente buco bianco
        // fra il modulo e il pulsante. Se il modulo cresce (risultati della
        // ricerca, nuovo cliente, orari) la tendina lo segue fin sotto il notch,
        // poi scorre; trascinando non cambia altezza.
        .fittedBrandSheet(height: headerHeight + formHeight + footerHeight)
        .presentationCornerRadius(26)
        .onChange(of: viewModel.done) {
            if viewModel.done { onDismiss() }
        }
        .sheet(isPresented: $showDatePicker) {
            BrandDatePickerDialog(
                initial: viewModel.date,
                onDismiss: { showDatePicker = false },
                onConfirm: { date in
                    viewModel.selectDate(date)
                    showDatePicker = false
                }
            )
        }
    }

    private var clientList: some View {
        VStack(spacing: 0) {
            ForEach(viewModel.results.prefix(3)) { client in
                let selected = viewModel.selectedClient?.id == client.id
                Button {
                    viewModel.selectClient(client)
                } label: {
                    HStack(spacing: 11) {
                        Circle()
                            .fill(Color.bone)
                            .frame(width: 34, height: 34)
                            .overlay(
                                Text(client.initials)
                                    .font(Typo.cormorant(12, weight: .regular))
                                    .foregroundStyle(Color.ink)
                            )
                        VStack(alignment: .leading, spacing: 0) {
                            Text(client.fullName)
                                .font(Typo.titleSmall)
                                .foregroundStyle(Color.ink)
                                .lineLimit(1)
                            Text("\(client.phone) · \(L("staff_booking_visits", client.visitCount))")
                                .font(Typo.jost(12))
                                .foregroundStyle(Color.textMuted)
                                .lineLimit(1)
                        }
                        Spacer()
                    }
                    .padding(.horizontal, 14)
                    .padding(.vertical, 12)
                    .background(selected ? Color.oliveTint : Color.bone)
                }
                .buttonStyle(.plain)
            }
            Button(action: viewModel.startCreateClient) {
                HStack(spacing: 11) {
                    Circle()
                        .fill(Color.bone)
                        .frame(width: 34, height: 34)
                        .overlay(
                            Image(systemName: "plus")
                                .font(.system(size: 13))
                                .foregroundStyle(Color.oliveWood)
                        )
                    Text(L("staff_booking_new_client"))
                        .font(Typo.bodyMedium)
                        .foregroundStyle(Color.textMuted)
                    Spacer()
                }
                .padding(.horizontal, 14)
                .padding(.vertical, 12)
            }
            .buttonStyle(.plain)
        }
        .background(RoundedRectangle(cornerRadius: Radii.md).fill(Color.bone))
        .overlay(
            RoundedRectangle(cornerRadius: Radii.md)
                .strokeBorder(Color.stoneBorder, lineWidth: 1.5)
        )
        .clipShape(RoundedRectangle(cornerRadius: 16))
    }

}

File: android/feature/staff/src/main/kotlin/com/devora/mencare/feature/staff/StaffBookingScreen.kt
package com.devora.mencare.feature.staff

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.imePadding
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Add
import androidx.compose.material.icons.outlined.ArrowDropDown
import androidx.compose.material.icons.outlined.Search
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.devora.mencare.core.common.formatDateShort
import com.devora.mencare.core.common.formatDuration
import com.devora.mencare.core.common.formatTime
import com.devora.mencare.core.designsystem.component.AccentButton
import com.devora.mencare.core.designsystem.component.BarAction
import com.devora.mencare.core.designsystem.component.BrandDatePickerDialog
import com.devora.mencare.core.designsystem.component.BrandSectionLabel
import com.devora.mencare.core.designsystem.component.BrandSwitch
import com.devora.mencare.core.designsystem.component.DropdownOption
import com.devora.mencare.core.designsystem.component.FilledTextField
import com.devora.mencare.core.designsystem.component.MultiSelectDropdown
import com.devora.mencare.core.designsystem.component.NameField
import com.devora.mencare.core.designsystem.component.PhoneField
import com.devora.mencare.core.designsystem.component.SlotChipRow
import com.devora.mencare.core.designsystem.theme.Bone
import com.devora.mencare.core.designsystem.theme.Cormorant
import com.devora.mencare.core.designsystem.theme.ErrorRed
import com.devora.mencare.core.designsystem.theme.Ink
import com.devora.mencare.core.designsystem.theme.OliveLight
import com.devora.mencare.core.designsystem.theme.OliveWood
import com.devora.mencare.core.designsystem.theme.Stone
import com.devora.mencare.core.designsystem.theme.TextMuted
import com.devora.mencare.core.ui.crm.FixedClientRow
import java.time.LocalDate
import java.time.LocalTime

/** Staff self-service booking sheet: same flow as the admin manual booking, but the
 * operator is always "me" — no operator picker, services limited to what this
 * operator performs. Lets staff take a phone/walk-in booking without the titolare.
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun StaffBookingScreen(
    onDismiss: () -> Unit,
    date: LocalDate = LocalDate.now(),
    time: LocalTime? = null,
    // "Modifica" dal dettaglio appuntamento: modulo precompilato con questo appuntamento.
    editAppointment: com.devora.mencare.core.model.Appointment? = null,
    editClient: com.devora.mencare.core.model.ClientRecord? = null,
    // Una chiave per apertura: ogni sheet riparte da un modulo vuoto.
    sessionKey: String = "staff-booking",
    viewModel: StaffBookingViewModel = hiltViewModel(key = sessionKey),
) {
    val state by viewModel.state.collectAsStateWithLifecycle()
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
    var showDatePicker by rememberSaveable { mutableStateOf(false) }

    LaunchedEffect(viewModel) {
        if (editAppointment != null) {
            viewModel.applyEdit(editAppointment, editClient)
        } else {
            viewModel.applyPreset(date, time)
        }
    }

    LaunchedEffect(state.done) {
        if (state.done) onDismiss()
    }

    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState,
        containerColor = Bone,
        shape = RoundedCornerShape(topStart = 26.dp, topEnd = 26.dp),
    ) {
        Column(Modifier.navigationBarsPadding()) {
            Row(
                modifier = Modifier.fillMaxWidth().padding(horizontal = 20.dp, vertical = 6.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Box(Modifier.width(72.dp), contentAlignment = Alignment.CenterStart) {
                    BarAction(stringResource(R.string.staff_booking_cancel), onDismiss, color = Ink)
                }
                Text(
                    stringResource(
                        if (state.editingId != null) R.string.staff_booking_title_edit else R.string.staff_booking_title,
                    ),
                    fontFamily = Cormorant,
                    fontSize = 21.sp,
                    color = Ink,
                    textAlign = TextAlign.Center,
                    maxLines = 1,
                    modifier = Modifier.weight(1f),
                )
                Box(Modifier.width(72.dp), contentAlignment = Alignment.CenterEnd) {
                    BarAction(stringResource(R.string.staff_booking_save), viewModel::save, color = OliveLight)
                }
            }

            Column(
                modifier = Modifier
                    .weight(1f, fill = false)
                    .verticalScroll(rememberScrollState())
                    .imePadding()
                    .padding(horizontal = 20.dp)
                    .padding(top = 12.dp, bottom = 12.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp),
            ) {
                BrandSectionLabel(stringResource(R.string.staff_booking_client))
                // In modifica il cliente è quello dell'appuntamento: si mostra e basta.
                if (state.editingId != null) {
                    FixedClientRow(state.selectedClient)
                } else {
                FilledTextField(
                    value = state.query,
                    onValueChange = viewModel::search,
                    label = "",
                    placeholder = stringResource(R.string.staff_booking_search_hint),
                    leadingIcon = Icons.Outlined.Search,
                )
                Column(
                    modifier = Modifier.fillMaxWidth().clip(RoundedCornerShape(16.dp)).background(Stone),
                ) {
                    state.results.take(3).forEach { client ->
                        val selected = state.selectedClient?.id == client.id
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .background(if (selected) OliveWood.copy(alpha = 0.18f) else Stone)
                                .clickable { viewModel.selectClient(client) }
                                .padding(horizontal = 14.dp, vertical = 12.dp),
                            verticalAlignment = Alignment.CenterVertically,
                        ) {
                            Box(
                                modifier = Modifier.size(34.dp).clip(CircleShape).background(Bone),
                                contentAlignment = Alignment.Center,
                            ) {
                                Text(
                                    "${client.firstName.first()}${client.lastName.first()}",
                                    fontFamily = Cormorant,
                                    fontSize = 12.sp,
                                    color = Ink,
                                )
                            }
                            Column(Modifier.weight(1f).padding(start = 11.dp)) {
                                Text(
                                    "${client.firstName} ${client.lastName}",
                                    style = MaterialTheme.typography.titleSmall,
                                    color = Ink,
                                    maxLines = 1,
                                    overflow = TextOverflow.Ellipsis,
                                )
                                Text(
                                    "${client.phone} · ${
                                        stringResource(R.string.staff_booking_visits, client.visitCount)
                                    }",
                                    style = MaterialTheme.typography.bodySmall.copy(fontSize = 12.sp),
                                    color = TextMuted,
                                    maxLines = 1,
                                    overflow = TextOverflow.Ellipsis,
                                )
                            }
                        }
                    }
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .clickable(onClick = viewModel::startCreateClient)
                            .padding(horizontal = 14.dp, vertical = 12.dp),
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        Box(
                            modifier = Modifier.size(34.dp).clip(CircleShape).background(Bone),
                            contentAlignment = Alignment.Center,
                        ) {
                            Icon(
                                Icons.Outlined.Add,
                                contentDescription = null,
                                tint = OliveWood,
                                modifier = Modifier.size(16.dp),
                            )
                        }
                        Text(
                            stringResource(R.string.staff_booking_new_client),
                            style = MaterialTheme.typography.bodyMedium,
                            color = TextMuted,
                            modifier = Modifier.padding(start = 11.dp),
                        )
                    }
                }
                }

                if (state.creatingClient) {
                    Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                        NameField(
                            value = state.newFirst,
                            onValueChange = { viewModel.setNewClientField("first", it) },
                            label = stringResource(R.string.staff_booking_new_client_first),
                            error = state.newFirstError,
                            modifier = Modifier.weight(1f),
                        )
                        NameField(
                            value = state.newLast,
                            onValueChange = { viewModel.setNewClientField("last", it) },
                            label = stringResource(R.string.staff_booking_new_client_last),
                            error = state.newLastError,
                            modifier = Modifier.weight(1f),
                        )
                    }
                    PhoneField(
                        value = state.newPhone,
                        onValueChange = { viewModel.setNewClientField("phone", it) },
                        label = stringResource(R.string.staff_booking_new_client_phone),
                        error = state.newPhoneError,
                    )
                    if (state.newClientFailed) {
                        Text(
                            stringResource(R.string.staff_booking_error_new_client),
                            style = MaterialTheme.typography.bodySmall,
                            color = ErrorRed,
                        )
                    }
                    AccentButton(
                        text = stringResource(R.string.staff_booking_new_client),
                        onClick = viewModel::createClient,
                        loading = state.creatingClientBusy,
                        height = 48.dp,
                        shape = RoundedCornerShape(16.dp),
                    )
                }

                Column {
                    BrandSectionLabel(stringResource(R.string.staff_booking_date))
                    Spacer(Modifier.height(8.dp))
                    DateField(value = formatDateShort(state.date)) {
                        showDatePicker = true
                    }
                }

                BrandSectionLabel(stringResource(R.string.staff_booking_services))
                // Stessa tendina multi-selezione della prenotazione del titolare.
                MultiSelectDropdown(
                    options = state.services.map { DropdownOption(it.id, it.name) },
                    selectedIds = state.selectedServiceIds,
                    onToggle = viewModel::toggleService,
                    placeholder = stringResource(R.string.manual_services_placeholder_staff),
                )

                BrandSectionLabel(
                    stringResource(R.string.staff_booking_slots, formatDuration(state.totalMinutes)),
                )
                SlotChipRow(
                    slots = state.slots,
                    selected = state.selectedSlot,
                    onSelect = viewModel::selectSlot,
                    label = ::formatTime,
                    emptyLabel = state.preferredSlot
                        ?.let { stringResource(R.string.staff_booking_slots_empty_preferred, formatTime(it)) }
                        ?: stringResource(R.string.staff_booking_slots_empty),
                )
                val preferred = state.preferredSlot
                if (state.preferredUnavailable && preferred != null) {
                    Text(
                        stringResource(R.string.staff_booking_slot_unavailable, formatTime(preferred)),
                        style = MaterialTheme.typography.bodySmall.copy(fontSize = 12.sp),
                        color = TextMuted,
                    )
                }

                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(16.dp))
                        .background(Stone)
                        .padding(horizontal = 16.dp, vertical = 12.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Column(Modifier.weight(1f)) {
                        Text(
                            stringResource(R.string.staff_booking_sms),
                            style = MaterialTheme.typography.titleMedium.copy(fontSize = 15.sp),
                            color = Ink,
                        )
                        Text(
                            stringResource(R.string.staff_booking_sms_hint),
                            style = MaterialTheme.typography.bodySmall.copy(fontSize = 12.sp),
                            color = TextMuted,
                        )
                    }
                    BrandSwitch(checked = state.sendSms, onCheckedChange = viewModel::setSendSms)
                }

                state.bookingError?.let { error ->
                    Text(
                        stringResource(
                            when (error) {
                                StaffBookingError.SLOT_TAKEN -> R.string.staff_booking_error_slot_taken
                                StaffBookingError.GENERIC -> R.string.staff_booking_error_save
                            },
                        ),
                        style = MaterialTheme.typography.bodySmall,
                        color = ErrorRed,
                    )
                }

                AccentButton(
                    text = stringResource(
                        if (state.editingId != null) R.string.staff_booking_cta_edit else R.string.staff_booking_cta,
                    ),
                    onClick = viewModel::save,
                    enabled = state.canSave,
                    loading = state.saving,
                    height = 56.dp,
                    shape = RoundedCornerShape(16.dp),
                )
            }
        }
    }

    if (showDatePicker) {
        BrandDatePickerDialog(
            initial = state.date,
            onDismiss = { showDatePicker = false },
            onConfirm = {
                viewModel.selectDate(it)
                showDatePicker = false
            },
        )
    }
}

/** Stone field that reads as a picker: value plus a chevron, as in the manual-booking sheet. */
@Composable
private fun DateField(value: String, onClick: () -> Unit) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .height(56.dp)
            .clip(RoundedCornerShape(16.dp))
            .background(Stone)
            .clickable(onClick = onClick)
            .padding(horizontal = 14.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Text(
            value,
            style = MaterialTheme.typography.bodyLarge,
            color = Ink,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis,
            modifier = Modifier.weight(1f),
        )
        Icon(
            Icons.Outlined.ArrowDropDown,
            contentDescription = null,
            tint = TextMuted,
            modifier = Modifier.size(20.dp),
        )
    }
}
